pubsub/gcppubsub/gcppubsub.go

```diff
@@ -392,14 +448,34 @@ type subscription struct {
 }
 
 // SubscriptionOptions will contain configuration for subscriptions.
-type SubscriptionOptions struct{}
+type SubscriptionOptions struct {
+	ReceiveMaxHandlers  int
```

Comment:
All of these that remain would need to be documented.

Some of them are not easy to explain because they are used in non-trivial ways by the concrete implementation.

I think we can keep only `MaxBatchSize` (no need for the `Receive` prefix), with the docstring something like:

"MaxBatchSize caps the maximum batch sized used when retrieving messages. It defaults to 1000."